packages/atomic/src/components/result-template-components/atomic-result-badge/atomic-result-badge.tsx

```diff
@@ -0,0 +1,79 @@
+import {Component, Prop, h} from '@stencil/core';
+
+/**
+ * The `atomic-result-badge` element renders a badge containing a field.
+ */
+@Component({
+  tag: 'atomic-result-badge',
+  styleUrl: 'atomic-result-badge.pcss',
+  shadow: true,
+})
+export class AtomicResultBadge {
+  /**
+   * The result field which the component should use.
+   * This will look in the Result object first, and then in the Result.raw object for the fields.
+   * It is important to include the necessary field in the ResultList component.
+   */
+  @Prop() public field?: string;
+
+  /**
+   * The text to display instead of the field.
+   */
+  @Prop() public text?: string;
+
+  /**
+   * Specifies the icon to display, either from the list of available icons or a direct link.
+   */
+  @Prop() public icon?: string;
+
+  /**
+   * Specifies a color to use for the badge.
+   *
+   * Defaults to `neutral-light`.
+   */
+  @Prop() public color?: string;
```

Comment:
Why not leverage parts or css variables for styling?

Reply:
If we have so style properties leveraging those tools, but also have "inline" style in props like this, isnt it a bit confusing to users?

Reply:
I'll add parts so badges can be customized and variables so their default color can be changed as well. Looking back, I no longer think the prop is necessary either since style can be added directly in the result template and classes/ids can be given to individual badges.